poisson_lib.py

```
@@ -0,0 +1,716 @@
+        result = model.fit(disp=0)
+        
+        # Calculate fit statistics
+        try:
+            stats_metrics = {
+                "deviance": result.deviance,
+                "pearson_chi2": result.pearson_chi2,
+                "aic": result.aic,
+                "bic": result.bic
+            }
+        except (AttributeError, ZeroDivisionError) as e:
+            logger.debug(f"Failed to calculate Poisson fit stats: {e}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)